### DISCUSSION TITLE
Use touchscreen during boot (in Plymouth)

### DISCUSSION THREAD

**Discussion by zoepzaroep** [linux-surface/linux-surface#844] [Q&A]
**Background:**

If you have an encrypted filesystem you are forced to enter the decryption-password via a physical keyboard. If it is somehow possible to make the touchscreen work in plymouth (during boot), you could hack together a little on-screen-keyboard. This has to be possible somehow (since rEFInd has touch support as well), but somehow I couldn't get it to work yet.

**TLDR:**

I am trying to use the touchscreen (on my Surface Pro 6 with the linux-surface kernel installed) during the boot splash screen which is provided by plymouth.

**What I did try already:**

I added the `ipts` and the `hid_multitouch` module to the `/etc/initramfs-tools/modules`.

But, for the touchscreen to work, I guess the `iptsd.service` has to be loaded as well.

So I tried to start the `iptsd.service` together with the `plymouth-start.service` by editting the `/usr/lib/systemd/system/plymouth-start.service` and adding the `iptsd.service` to the `After` section:

```
[Unit]
Description=Show Plymouth Boot Screen Default
Dependencies=no
Wants=systemd-ask-password-plymouth.path
After=systemd-udev-trigger.service systemd-udevd.service keyboard-setup.service iptsd.service Before=systemd-ask-password-plymouth.service
ConditionKernelCommandLine=!plymouth.enable=0
ConditionKernelCommandLine=!nosplash
ConditionKernelCommandLine=splash
ConditionVirtualization=!container
IgnoreOnIsolate=true

[Service]
ExecStart=/sbin/plymouthd --mode=boot --pid-file=/run/plymouth/pid --attach-to-session
ExecStartPost=-/bin/plymouth show-splash
Type=forking
RemainAfterExit=yes
KillMode=mixed
SendSIGKILL=no
```

But this does not seems to be all. Does somebody else have an idea how to get the touchscreen working in plymouth?

**Comment by qzed** [comment]
IIRC, IPTSd requires uinput, so you'll need to add that to the initrram as well. In addition you also need at least the `mei` module (not sure if that's all though...).

**Comment by zoepzaroep** [comment]
Thanks for your help that makes sense. I added the following modules to `/etc/initramfs-tools/modules`:

```
hid_multitouch
ipts
uinput
mei
mei_hdcp
mei_me
```

Still does not work.

**Comment by StollD** [comment]
Plymouth does not have an onscreen keyboard: https://gitlab.freedesktop.org/plymouth/plymouth/-/issues/144
